SDM generator

jessitron opened the pull request:
this adds a new generator based on github-sdm.

The bit about finding the author doesn't work, graphclient returns a 400, but that's a different problem.

jessitron committed:
upgrade client

jessitron committed:
accurately name generator

jessitron committed:
Add sdm generator

and update more info in package.json

Add SDM generator

jessitron committed:
Automatic de-linting
[atomist:auto-delint]

jessitron committed:
Remove some unused stuff that breaks compilation on travis

but not here, here it seems not to care

ddgenome reviewed (changes requested):
The logzio stuff is very much used.

ddgenome commented:
This should fix your logzio compilation problems: https://github.com/atomist/k8-automation/blob/3a0c01eb218cccbe18e42e7de5a90f934949666a/src/index.d.ts#L43

jessitron committed:
now the author part works

jessitron committed:
Revert "Remove some unused stuff that breaks compilation on travis"

This reverts commit 80fa76a61f8bef9dbb636fee338212af21561547.

Commit:
Update logzio, package.json

Update to latest version of logzio.ts to get things compiling again.
Update package scripts.  Remove generated typings.

Commit:
Downgrade typedoc

ddgenome commented on config/default.json:
```diff
@@ -1,10 +1,10 @@
-    "api":"https://automation-staging.atomist.services/registration",
-    "graphql": "https://automation-staging.atomist.services/graphql/team"
+    "api": "https://automation.atomist.com/registration",
+    "graphql": "https://automation.atomist.com/graphql/team"
```
These changes will break the staging.conf. Did you mean to commit these? I typically create a new conf and use that to test in a specific team.

ddgenome commented on config/default.json:
```diff
@@ -1,10 +1,10 @@
   },
   "groups": null,
-  "teamIds": "T1L0VDKJP",
+  "teamIds": "T1JVCMVH7",
```
Intentional?

jessitron replied:
nope, that's satellite-of-love. better to leave it as community

ddgenome commented on src/commands/editor/node/updatePackageLock.ts:
```diff
@@ -0,0 +1,22 @@
+import * as child_process from "child_process";
+
+export async function updatePackageLock(project: Project) {
+    const result = new Promise((resolve, reject) => child_process.exec("npm install",
```
This seems like a lot of compute/network just to update the package-lock.json. Couldn't we just update the package name and version manually?

jessitron replied:
did it have a package-lock before? 
... yeah, I guess it does
is it accurate?
... maybe

so, probably. But the way I update package-lock.json is by running npm install, so I wanted the automation to do that.

It also has the effect of applying formatting to package.json, which masks a bug in client that messed up the spacing when we changed that.

jessitron replied:
(I fixed the bug in client but it isn't released afaik)

ddgenome reviewed (commented):
A couple questions.
